server/base/src/main/java/org/apache/accumulo/server/util/MetadataTableUtil.java

```diff
@@ -894,12 +894,19 @@ public static void removeBulkLoadEntries(Connector conn, String tableId, long ti
         BatchWriter bw = conn.createBatchWriter(MetadataTable.NAME, new BatchWriterConfig())) {
       mscanner.setRange(new KeyExtent(tableId, null, null).toMetadataRange());
       mscanner.fetchColumnFamily(TabletsSection.BulkFileColumnFamily.NAME);
+      boolean shouldTrace = log.isTraceEnabled();
+      String tidString = Long.toString(tid);
       for (Entry<Key,Value> entry : mscanner) {
-        log.debug("Looking at entry " + entry + " with tid " + tid);
-        if (Long.parseLong(entry.getValue().toString()) == tid) {
-          log.debug("deleting entry " + entry);
-          Mutation m = new Mutation(entry.getKey().getRow());
-          m.putDelete(entry.getKey().getColumnFamily(), entry.getKey().getColumnQualifier());
+        if (shouldTrace) {
+          log.trace("Looking at entry " + entry + " with tid " + tidString);
```

Comment:
Yes it was related to the forced logger:
org.apache.log4j.Category.callAppenders(Category:204)
waiting for lock .. (a org.apache.log4j.Logger)
..Category.forcedLog..
..Category.debug..
... removeBulkEntries(MetadataTableUtil.java:911)

Yes, it looks like the slf4j implementation I checked avoids the forceLog call and instead checks against the level which would not block.